php/class-customize-concurrency.php

```diff
@@ -14,4 +14,204 @@
  */
 class Customize_Concurrency {
 
-}
+	/**
+	 * Post type that stores settings changes until session ends.
+	 *
+	 * Limited by WP to 20 characters
+	 *
+	 * @var string $POST_TYPE
+	 */
+	const POST_TYPE = 'custom_saved_setting';
+
+	/**
+	 * Plugin instance.
+	 *
+	 * @var Plugin $plugin
+	 */
+	public $plugin;
+
+	/**
+	 * Customize manager.
+	 *
+	 * @var \WP_Customize_Manager
+	 */
+	public $customize_manager;
+
+	/**
+	 * Whether kses filters on content_save_pre are added.
+	 *
+	 * @var bool
+	 */
+	protected $kses_suspended = false;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Plugin $plugin Plugin instance.
+	 */
+	public function __construct( Plugin $plugin ) {
+		$this->plugin = $plugin;
+		$this->customize_manager = null;
+
+		// Bootstrap the Customizer.
+		if ( ! empty( $GLOBALS['wp_customize'] ) ) {
+			$this->customize_manager = $GLOBALS['wp_customize'];
+		}
+
+		add_action( 'init', array( $this, 'register_post_type' ) );
+		add_action( 'customize_controls_enqueue_scripts', array( $this, 'customize_controls_enqueue_scripts' ) );
+		add_action( 'customize_controls_print_footer_scripts', array( $this, 'customize_controls_print_footer_scripts' ) );
+		add_filter( 'wp_insert_post_data', array( $this, 'preserve_inserted_post_name' ), 10, 2 );
+		add_action( 'customize_save', array( $this, 'customize_save' ), 1000 );
+	}
+
+	/**
+	 * Register the post type.
+	 */
+	public function register_post_type() {
+		$args = array(
+			'labels' => array(
+				'name' => __( 'Customize Saved Setting', 'customize-concurrency' ),
+			),
+			'description' => __( 'Copy of Customizer setting value for tracking concurrency.', 'customize-concurrency' ),
+			'public' => false,
+			'hierarchical' => false,
+		);
+		register_post_type( self::POST_TYPE, $args );
+	}
+
+	/**
+	 * Add scripts & styles.
+	 *
+	 * @action customize_controls_enqueue_scripts
+	 */
+	public function customize_controls_enqueue_scripts() {
+		wp_enqueue_script( 'wp-customize-concurrency' );
+	}
+
+	/**
+	 * Put timestamp and footer in hidden field.
+	 *
+	 * todo: find a better way to get this into what is sent via ajax
+	 */
+	public function customize_controls_print_footer_scripts() {
+		printf( '<input type="hidden" id="customizer_session_timestamp" value="%s" />', time() );
+		printf( '<input type="hidden" id="current_user_id" value="%s" />', get_current_user_id() );
```

Comment:
I don't think that outputting these inputs will work as expected. You'll need to export a new JS object that contains these values, and then you can inject them into the Customizer requests like so: https://github.com/xwp/wp-customize-snapshots/blob/68e8f34d5bcb8c73307eaa2e91e19824396251a5/js/customize-snapshots.js#L122-L137